Alert/doc/spec.md

```diff
@@ -0,0 +1,155 @@
+direction.
+
+Some examples of context data are: type of alert (traffic, weather, security,
+and pollution, etc.), severity, location and so on.
+
+## Data model
+
+A JSON Schema corresponding to this data model can be
+found [here](../schema.json).
+
++ `id` : Entity's unique identifier.
+
++ `type` : Entity type. It must be equal to `Alert`.
+
++ `category` : Define the category of alert (Traffic jam, accidents, weather
+  conditions, high level of pollutants)
+  + Attribute type: [Text](https://schema.org/Text)
+  + Allowed values:
+    + (`traffic`, `weather`, `environment`, `health`, `security` )
+  + Mandatory
+
++ `subCategory` : Describe the sub category of alert.
+  + Attribute type: [Text](https://schema.org/Text)
+  + Allowed values:
+    + (`trafficJam`, `carAccident`, `carWrongDirection`, `carStopped`,
+      `pothole`, `roadClosed`, `roadWorks`, `hazardOnRoad`, `injuredBiker`)
+      (for `traffic` category)
+    + (`rainfall`, `highTemperature`, `lowTemperature`, `heatWave`, `ice`,
+      `snow`, `wind`, `fog`, `flood`, `tsunami`, `tornado`, `tropicalCyclone`,
+      `hurricane` ) (for `weather` category)
+    + (`airPollution`, `waterPollution`, `pollenConcentration`)
+      (for `environment` category)
+    + (`asthmaAttack`, `bumpedPatient`, `fallenPatient`, `heartAttack` )
+      (for `health` category)
+    + (`suspiciousAction`, `robbery`, `assault` ) (for `security` category)
+  + Optional
+
++ `location` : Location of alert represented by a GeoJSON geometry.
+  + Attribute type: `geo:json`.
+  + Normative References: [rfc7946](https://tools.ietf.org/html/rfc7946)
+  + Mandatory if `address` is not present
+
++ `address` : Civic address of `Alert`
+  + Attribute type: [Address](https://schema.org/address)
+  + Mandatory if `location` is not present.
+
++ `dateTimeObserved` : The date and time the observation of the alert
+  in ISO8601 UTC format.
+  + Attribute type: [DateTime](https://schema.org/DateTime).
+  + Normative References: [ISO8601](https://www.iso.org/standard/40874.html)
+  + Mandatory
+
++ `dateTimeValidity` : The date and time of validity of the alert in ISO8601
+  UTC format. It can be represented by an specific time instant (the time
+  when the alert expire) or by an ISO8601 interval.
+  + Attribute type: [DateTime](https://schema.org/DateTime) or
+    an ISO8601 interval.
+  + Normative References: [ISO8601](https://www.iso.org/standard/40874.html)
+  + Optional
+
++ `description` : A description of alert (e.g. Traffic jam in Paseo de la
+   Reforma. Emergency services at place).
+  + Attribute type: [Description](https://schema.org/description)
+  + Optional
+
++ `alertSource` : reference to the source of the alert. For example, it could be a user of an application, a device, or a service.
+  + Attribute type: [Text](https://schema.org/Text)
+    or [URL](https://schema.org/URL)
+  + Mandatory.
+
++ `data` : used to carry additional data for the alert.
```

Comment:
do we really need this?

Reply:
maybe we can rename it to additionalData, but in general having a way to add data related to specific alerts without having to encode them in the data model i think is useful.

for example, this alert model can be used for [GTFS Service Alerts](https://developers.google.com/transit/gtfs-realtime/examples/alerts), and allow to include easily additional fields such as:
```
    # cause of the alert - see gtfs-realtime.proto for valid values
    cause: CONSTRUCTION
    # effect of the alert - see gtfs-realtime.proto for valid values
    effect: DETOUR
```